Dockerfile.minimal

```diff
@@ -0,0 +1,27 @@
+FROM golang:1.7-alpine AS builder 
+
+ENV DISTRIBUTION_DIR /go/src/github.com/docker/distribution
+ENV DOCKER_BUILDTAGS include_oss include_gcs
+
+RUN set -ex \
+    && apk add --no-cache make git gcc libc-dev
+
+WORKDIR $DISTRIBUTION_DIR
+COPY . $DISTRIBUTION_DIR
+
+RUN make PREFIX=/go clean binaries
```

Comment:
don't think `make clean` is needed in context of the Dockerfile (assuming we just add the `bin` directory to `.dockerignore`)

also wondering if we should build directly, instead of using `make` here (but haven't looked closely into that)